### ISSUE TITLE
Grid exportOptions should support a "rows" property to allow some customization of which rows are exported.

### ISSUE THREAD

**Issue by cnrudd** [xh/hoist-react#2419]
The "rows" property should be similar to the "columns" property.
It should accept these string values:
"ALL", "FILTERED"

and accept a function taking the GridModel.  Not yet sure what that function will return maybe an array of records, or Id's, or a boolean like a filter...

**Comment by cnrudd** [comment]
I've looked at the code in [GridExportService.js](https://github.com/xh/hoist-react/blob/de9d34955e24d8e6ad68b58bed17ed4e9cc4298f/svc/GridExportService.js) and see that the gathering of cell data depends on the rows as rendered by agGrid. 

The service uses agrid's aggData to assemble summary rows.

If we were to apply a 2nd filter during export via the new `rows` option, I was hopefully thinking it would only be applied to the store records to create a new recordSet, without re-rendering the grid, but then the summary rows that might be left and are consulted for aggData could contain inaccurate values.

Basically, a rows option could work for plain grids, but for any grids that have aggData, it wouldn't work, or would be very difficult to implement, requiring a re-render of the grid using the export filter, and then a re-render back to the previous state.
Probably difficult to implement, creates UI jiggles & wiggles, and not in great demand.

Will stop work on this ticket until discussed further.

**Comment by lbwexler** [comment]
Colin -- I am not sure I get this -- I was thinking that if you choose to filter your grid export, its up to you to consistently deal with your aggregates.    Its really use at your own risk.  We aren't making any promises about recomputing anything.

Having said that, if you don't need this anymore and have solved the problem that led to the ticket it in user land, I would be happy to let this one go.
